lib/ipam/ipam.go

```diff
@@ -686,6 +735,7 @@ func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, a
 				// Grab a new random block.
 				blockCIDR := newBlockCIDR()
 				if blockCIDR == nil {
+					iPAMAssignmentInfo.ExhaustedPools = append(iPAMAssignmentInfo.ExhaustedPools, p.Spec.CIDR)
```

Comment:
Maybe nothing useful can be done about this, and it's a very rare and obscure case, so no big deal. Best thing we could do would perhaps be to include a disclaimer - "IP pool is exhaused (but btw we hit an error just before this)"

Reply:
To be completely honest, I was led to this by the debug message on line 739 ("All addresses exhausted in pool %s"). But I'm not sure I follow, does `randomBlockGenerator()` keep track of all the blocks it returned (so that if we skip it due to data store query failure, it will not ever be generated again)?